include/zephyr/data/json.h

```diff
@@ -106,6 +106,9 @@ struct json_obj_descr {
 	/* 65535 bytes is more than enough for many JSON payloads. */
 	uint32_t offset : 16;
 
+	/* Size of the structure being described by the "object" descriptor */
+	size_t struct_size;
```

Comment:
Remove the `align_shift` member. This frees up 2 bits, these can be used to `field_name_len` and `type` to get an extra bit?